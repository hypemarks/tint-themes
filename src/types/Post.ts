import { IPost, Nullable } from '@tintup/tint-sdk';

export interface ICustomCTA {
  text: string;
  url: string;
  thumbnail_url: Nullable<string>;
  type: string;
  coordinates: Nullable<string[]>;
  id: string;
}

export interface IPostWithCTA extends IPost {
  cta: ICustomCTA[];
}
