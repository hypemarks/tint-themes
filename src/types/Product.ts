import { IProductTags, ITaggedProduct } from '@tintup/tint-sdk/lib';

export { IProductTags, ITaggedProduct };
