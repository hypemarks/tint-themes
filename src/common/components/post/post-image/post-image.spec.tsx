import React from 'react';
import { render } from '@testing-library/react';
import { buildPost } from '../../../../../test/generate';
import '../../../../../test/intersectionObserverMock';
import { IPostWithCTA } from '../../../../types/Post';
import PostImage from './post-image';

describe('Test <PostImage /> Component', () => {
  const post = buildPost({
    cta: [],
    attributes: {
      imageUrl: 'image_url',
    },
  });

  it('should match snapshot', () => {
    const wrapper = render(<PostImage post={post as IPostWithCTA} onClick={jest.fn} onCtaClick={jest.fn} />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
