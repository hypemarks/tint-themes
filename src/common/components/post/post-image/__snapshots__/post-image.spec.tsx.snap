// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <PostImage /> Component should match snapshot 1`] = `
<DocumentFragment>
  <a
    class="post-image-wrapper--no-image"
    data-testid="post-image-wrapper-anchor"
    rel="noopener noreferrer"
    target="_blank"
  />
</DocumentFragment>
`;
